ddl/ddl_tiflash_api.go

```diff
@@ -397,53 +535,44 @@ func (d *ddl) pollTiFlashReplicaStatus(ctx sessionctx.Context, pollTiFlashContex
 		})
 		// We only check unavailable tables here, so doesn't include blocked add partition case.
 		if !available {
-			allReplicaReady = false
 			enabled, inqueue, _ := pollTiFlashContext.Backoff.Tick(tb.ID)
 			if inqueue && !enabled {
 				logutil.BgLogger().Info("Escape checking available status due to backoff", zap.Int64("tableId", tb.ID))
 				continue
 			}
 
-			// We don't need to set accelerate schedule for this table, since it is already done in DDL, when
-			// 1. Add partition
-			// 2. Set TiFlash replica
-
-			// Compute sync data process by request TiFlash.
-			regionReplica := make(map[int64]int)
-			for _, store := range pollTiFlashContext.TiFlashStores {
-				err := helper.CollectTiFlashStatus(store.Store.StatusAddress, tb.ID, &regionReplica)
+			progress, err := getTiFlashTableSyncProgress(pollTiFlashContext, tb.ID, tb.Count)
+			if err != nil {
+				logutil.BgLogger().Error("get tiflash sync progress failed",
+					zap.Error(err),
+					zap.Int64("tableID", tb.ID),
+				)
+				continue
+			}
+			if pollTiFlashContext.ProgressCache[tb.ID] != progress {
+				err = infosync.UpdateTiFlashTableSyncProgress(context.Background(), tb.ID, progress)
 				if err != nil {
-					return allReplicaReady, errors.Trace(err)
+					logutil.BgLogger().Error("updating TiFlash replica process failed",
+						zap.Error(err),
+						zap.Int64("tableID", tb.ID),
+						zap.String("progress", progress),
+					)
+					continue
 				}
+				pollTiFlashContext.ProgressCache[tb.ID] = progress
 			}
 
-			logutil.BgLogger().Debug("CollectTiFlashStatus", zap.Any("regionReplica", regionReplica), zap.Int64("tableID", tb.ID))
-
-			var stats helper.PDRegionStats
-			if err := infosync.GetTiFlashPDRegionRecordStats(context.Background(), tb.ID, &stats); err != nil {
-				return allReplicaReady, err
-			}
-			regionCount := stats.Count
-			flashRegionCount := len(regionReplica)
-			avail := regionCount == flashRegionCount
+			avail := progress[0] == '1'
```

Comment:
Before, a table is available when every region has at least ONE replica. However, we now have a more strict criterion that all region must have a number of "catching-up" peers which is larger or equal to `SET REPLICA N`. So seems a table can have a larger chance to be unavailable.
So when a table is not available, we can only got a progress here, which give us no clue which peer of which region is not ok.
Is it possible and reasonable that we can print out which region peer is broken, when a table took too long to be ready?

Reply:
I will add logs in TiFlash.

Reply:
Is there any specific advantages to change to this more strict criterion? IMO it would be always better if we allow user to use TiFlash as soon as possible, as long as the query would be fine.

Reply:
> Is there any specific advantages to change to this more strict criterion? IMO it would be always better if we allow user to use TiFlash as soon as possible, as long as the query would be fine.

In fact, the `available` is not accurate before, so the query may fail even `available` is `1`.